llvm/unittests/IR/RepoTicketMDTest.cpp

```diff
@@ -802,3 +774,431 @@ TEST_F(DoubleModule, MixedFrontendAndBackendHashGeneration) {
   EXPECT_NE(getTicket(Foo)->getDigest(), BarDigest)
       << "Functions of foo and bar should have the different digest";
 }
+
+//
+// The following four tests are targeted on checking that the function
+// definition order should not affect the function hash. Modules M0 and M1 have
+// the same call graph but have different function definition order.
+//
+//  IR forming a following call graph for M0 and M1.
+//      c
+//     / \
+//    v   v
+//    a   b
+// Functions 'a' 'b' and 'c' should have the same hashes in Module M0 and M1.
+//
+TEST_F(DoubleModule, Simple) {
+  const char *Module0String = "define i32 @A(){\n"
+                              "entry:\n"
+                              "  ret i32 1\n"
+                              "}\n"
+                              "define i32 @B(){\n"
+                              "entry:\n"
+                              "  ret i32 2\n"
+                              "}\n"
+                              "define i32 @C() {\n"
+                              "entry:\n"
+                              "  %call = call i32 @A()\n"
+                              "  %call1 = call i32 @B()\n"
+                              "  %add = add nsw i32 %call, %call1\n"
+                              "  ret i32 %add\n"
+                              "}\n";
+  const char *Module1String = "define i32 @C() {\n"
+                              "entry:\n"
+                              "  %call = call i32 @A()\n"
+                              "  %call1 = call i32 @B()\n"
+                              "  %add = add nsw i32 %call, %call1\n"
+                              "  ret i32 %add\n"
+                              "}\n"
+                              "define i32 @B(){\n"
+                              "entry:\n"
+                              "  ret i32 2\n"
+                              "}\n"
+                              "define i32 @A(){\n"
+                              "entry:\n"
+                              "  ret i32 1\n"
+                              "}\n";
+
+  M0 = parseAssembly(Module0String);
```

Comment:
It looks like this is parsing IR.

Reply:
Yes, it is parsing the IR code. However, I used the `parseAssembly` as the function name which is also used by all the other unitttests.